core/src/test/java/org/jboss/jandex/test/BasicTestCase.java

```diff
@@ -468,6 +474,23 @@ public void testNullClass() throws IOException {
         indexer.indexClass(null);
     }
 
+    @Test
+    public void testRuntimeInvisibleAnnotations() throws IOException {
+        @RuntimeInvisible(placement = "class")
+        class Target {
+            @SuppressWarnings("unused")
+            void execute(@RuntimeInvisible(placement = "arg") String arg) {
+            }
+        }
+
+        Index index = Index.of(Target.class);
+        DotName annoName = DotName.createSimple(RuntimeInvisible.class.getName());
+        List<AnnotationInstance> rtInvisible = index.getAnnotations(annoName);
+        assertEquals(2, rtInvisible.size());
+        assertEquals(0, Target.class.getDeclaredAnnotations().length);
+        assertEquals(0, Target.class.getDeclaredMethods()[0].getDeclaredAnnotations().length);
```

Comment:
Pretty sure you want to assert against the `rtInvisible` data, not reflection data :-)

Reply:
With this one, it was a sanity check that the two invisible annotations from the class and method param were invisible for reflection. Would it be better to remove that?

Reply:
Ah gotcha. No that can stay, though I'd probably move it to the beginning, before creating the index. And I'd at least assert that the `AnnotationInstance`s correctly report that they're runtime-invisible.

Reply:
I think this should be good now. Adjusted the test to serialize/deserialize to version 11 and added another one for version 10.